### DISCUSSION TITLE
How to reload when code changes during development

### DISCUSSION THREAD

**Discussion by SamEdwardes** [RealOrangeOne/django-tasks#124] [Q&A] [answered]
Thank you for the great work on django-tasks. I am new to django, and have found your library really nice to work with. Is there a way to automatically reload django-tasks when my code changes? For example, I start the main process with:

```bash
uv run manage.py runserver
```

When code changes, it auto reloads:

```bash
2024-12-08 06:50:04.683 | WARNING  | config.settings:<module>:54 - {'message': 'DEBUG is True'}
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
December 08, 2024 - 06:50:05
Django version 5.1.2, using settings 'config.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.

[08/Dec/2024 06:50:05] "GET /email-screener HTTP/1.1" 200 624770
/Users/samedwardes/projects/personal/gmail-screener/gmailscreener/app_screener/models.py changed, reloading.
2024-12-08 06:50:15.764 | DEBUG    | config.settings:<module>:35 - App is starting!!!
```

Is there a way to do something similar with django-tasks? Here is how I start the process:

```bash
uv run manage.py db_worker
```

I am using the database backend.

```python
TASKS = {"default": {"BACKEND": "django_tasks.backends.database.DatabaseBackend"}}
```

I think this is related to https://github.com/RealOrangeOne/django-tasks/issues/105. Is there any work around I can do before this is implemented at the library level?

**Comment by RealOrangeOne** [comment]
Nothing first-party, no. You'll need to use an external tool to reload it.

See open issue: https://github.com/RealOrangeOne/django-tasks/issues/105

**Reply by SamEdwardes** [reply]
Thank you! Do you have any suggestions on tools that could do this?

**Reply by RealOrangeOne** [reply, marked as answer]
I quite like https://github.com/watchexec/watchexec

**Reply by SamEdwardes** [reply]
Thank you! This does the trick:

```bash
watchexec -r uv run manage.py db_worker
```

I looked at a few other options but could not get them to work:

- fswatch
- watchman

Thank you for your help.
